File: 2024/Leetcode/problem2405.py
class Solution:
    def partitionString(self, s: str) -> int:
        mySet = set()
        result = 1
        for char in s:
            if char not in mySet:
                mySet.add(char)
            else:
                mySet = set(char)
                result += 1

        return result

File: 2024/Leetcode/problem2405.cpp
class Solution {
public:
    int partitionString(string s) {
        int result = 1;
        unordered_set<char> substring;

        for (auto character: s) {
            if (substring.find(character) == substring.end()) {
                substring.insert(character);
            } else {
                substring.clear();
                substring.insert(character);
                result++;
            }
        }
        return result;
    }
};